src/network/blockchain.js

```diff
@@ -42,81 +43,61 @@ function createWeb3Instance({blockchainUrl, privateKey}) {
     return web3;
 }
 
-const abisByContractName = {};
-class Web3Bridge {
-    constructor(ctx) {
-        this.ctx = ctx;
-        this.connectionString = config.get('network.web3');
-        this.address = getNetworkAddress();
-        this.web3_call_retry_limit = config.get('network.web3_call_retry_limit');
-        this.web3 = this.ctx.web3 = this.ctx.network.web3 = this.createWeb3Instance(); // todo: maybe you should hide it behind this abstraction, no?
-        log.debug('Successfully created a web3 instance');
-        this.ctx.web3bridge = this;
-        this.start();
-    }
-
-    createWeb3Instance() {
-        return createWeb3Instance({
-            blockchainUrl: config.get('network.web3'),
-            privateKey: '0x' + getNetworkPrivateKey()
-        });
-    }
+function createBlockchain(basepath) {
+    const abisByContractName = {};
+    const web3CallRetryLimit = config.get('network.web3_call_retry_limit');
+    const web3 = createWeb3Instance({
+        blockchainUrl: config.get('network.web3'),
+        privateKey: '0x' + getNetworkPrivateKey()
+    });
 
-    async start() {}
+    const blockchain = {};
 
-    async loadPointContract(contractName, at) {
+    blockchain.loadPointContract = async (contractName, at) => {
         if (!(contractName in abisByContractName)) {
-            const buildDirPath = path.resolve(
-                resolveHome(config.get('datadir')),
-                'contracts'
-            );
+            const buildDirPath = path.resolve(resolveHome(config.get('datadir')), 'contracts');
 
             const abiFileName = path.resolve(buildDirPath, contractName + '.json');
 
             if (!fs.existsSync(abiFileName)) {
                 if (!fs.existsSync(buildDirPath)) {
                     fs.mkdirSync(buildDirPath, {recursive: true});
                 }
-                const contractPath = path.resolve(
-                    this.ctx.basepath,
-                    '..',
-                    'truffle',
-                    'contracts'
-                );
-                await compileContract({name: contractName, contractPath, buildDirPath});
+                const contractPath = path.resolve(basepath, '..', 'truffle', 'contracts');
```

Comment:
This is the only part where we use the `basepath`, but because of it we have to wrap the entire module into the `createBlockchain` function.

Let's make the `basepath` an argument to the `loadPointContract` function instead. It should also have a default value. Let's also compute it here so that it would be the same as the one we pass to the `createBlockchain` function.

The goal is to get rid of `createBlockchain` function completely. Instead we should simply export all the `blockchain` object methods, and each method should use a locally created `web3` closure.